### DISCUSSION TITLE
SafeEventEmitterProvider for react Jsx format

### DISCUSSION THREAD

**Discussion by sundeep138** [Web3Auth/Web3Auth#1383] [Help]
Hi there

Can anybody help us as we understand that SafeEventEmitterProvider may be coded in typescript.
Our dev  team are using react that is predominatly in jsx files.

Now the problem statement is that it does not compile when using SafeEventEmitterProvider in app.jsx. 
So got stuck. anywork around or pointers to resolving this issue will be great to learn and resolve our issue.

Thanks
Sundeep

**Comment by yashovardhan** [comment]
Can you share your compilation logs? And also your code related to it?

**Reply by sandippa26** [reply]
Hello @yashovardhan i'm facing an issue when using web3auth and auth0 it will give error "WalletLoginError: Failed to connect with wallet. Failed to login with openlogin"  
import { Web3AuthCore } from "@web3auth/core";
import {
  WALLET_ADAPTERS,
  CHAIN_NAMESPACES,
  SafeEventEmitterProvider
} from "@web3auth/base";
import { OpenloginAdapter } from "@web3auth/openlogin-adapter";
all this packages i'm using it cannot work on jsx files but if i was all this packages in typescript project tsx file it will work properly 
Thanks

**Reply by yashovardhan** [reply]
Can you please pass over these info in https://community.web3auth.io/t/safeeventemitterprovider-for-react-jsx-format/2925

We have moved our community platform to this website. Also please share the whole code, not just the packages related to web3auth.

**Comment by sundeep138** [comment]
Hi Prachi and Sandip

can you pls share those logs with the Web3 provider pls.
thanks
Sundeep

**Reply by sandippa26** [reply]
okay @sundeep138

**Comment by sundeep138** [comment]
thanks
Sundeep

**Comment by sundeep138** [comment]
Thanks Sandip.
Web3 team pls let us know what are we missing to make web3 modal work.
Thanks
Sundeep

**Comment by sundeep138** [comment]
Guys
Pls respond as we are eager to know the way forward.
Thanks
Sundeep

**Comment by yashovardhan** [comment]
Hey folks,

We have shifted our discussions to  community.web3auth.io

Can you please ask your questions there instead?

Here's a link to your imported thread for reference purposes. You can either reply there itself or create a new topic.

https://community.web3auth.io/t/safeeventemitterprovider-for-react-jsx-format/2925

**Comment by sundeep138** [comment]
Hi Web3 experts,

Pls let us know your response to our simple question whether you support
jsx when using safeemitter lib or give your number to call you all to
discuss.
Details of the log is also been supplied.
Thanks
Sundeep

**Comment by sundeep138** [comment]
Hi there
We are not getting a response from you all.
What should we do?.
Move on to another product or wait.
Let us know
Thanks
Sundeep

**Reply by sundeep138** [reply]
@yashovardhan can I get a phone number to talk to you all as a last resort.?

**Reply by yashovardhan** [reply]
Hey @sundeep138 

Please ask your question on https://community.web3auth.io

Please understand that our support is specific to that platform. 

I cannot share my number unfortunately.

**Comment by sundeep138** [comment]
What do you suggest I do? To Seek a response as we are awaiting since
couple of days.
Thanks
Sundeep

**Reply by yashovardhan** [reply]
As mentioned, please ask on https://community.web3auth.io, share your whole code related to web3auth. You'll get a proper resolution after that only. It is impossible for us to give you an answer with half information about how your code is structured. We have tried using Web3Auth in jsx and couldn't replicate your error.

**Comment by sundeep138** [comment]
Ok Yash. we will send you the whole code. We will appreciate if you can
give us a faster response.

I will be travelling back to Australia and whilst in India, I want this
issue to be resolved to use Web3 auth WITHOUT typescript.

Thanks
Sundeep

**Comment by sundeep138** [comment]
Hi yash

FYI our code for your review and let us know the workaround for not using
typescript.

Thanks
Sundeep
import React, { useState, useEffect } from 'react';
import { Container, Row, Col, Form } from 'react-bootstrap';
import {
  Button,
  Icon,
  TextField
} from ***@***.***/core';
import {
  ScrollingProvider,
  useScrollSection,
  Section,
} from 'react-scroll-section';

// import { WALLET_ADAPTERS, CHAIN_NAMESPACES, SafeEventEmitterProvider }
from ***@***.***/base";
import { Web3AuthCore } from ***@***.***/core";
// import { OpenloginAdapter } from ***@***.***/openlogin-adapter";
import walletConnectProvider from ***@***.***/web3-provider'
// import {EventEmitter} from 'events';
// import { SafeEventEmitterProvider } from
***@***.***/network/lib/react';
// import Web3Modal from 'web3modal';
// import { Web3Auth } from ***@***.***/modal";
import { Web3Auth } from ***@***.***/web3auth";
// import {Auth0LockPasswordless} from 'auth0-lock';
// import { Web3AuthCore } from ***@***.***/core";
import {
  WALLET_ADAPTERS,
  CHAIN_NAMESPACES,
  SafeEventEmitterProvider
} from ***@***.***/base";

import { OpenloginAdapter } from ***@***.***/openlogin-adapter";
import { Link } from 'react-router-dom';
// import { useAuth0 } from ***@***.***/auth0-react";
// import RPC from "../../services/web3RPC";

const clientId =

"BEglQSgt4cUWcj6SKRdu5QkOXTsePmMcusG5EAoyjyOYKlVRjIF1iCNnMOTfpzCiunHRrMui8TIwQPXdkQ8Yxuk";


const StaticMenu = (props) => {
  const [web3auth, setWeb3auth] = useState(null);
  const [provider, setProvider] = useState(null);
  const homeSection = useScrollSection('home');
  const pricingSection = useScrollSection('pricingTable');
  const helpSection = useScrollSection('help');

  useEffect(() => {
    const init = async () => {
      try {
        const web3auth = new Web3AuthCore({
          clientId,
          chainConfig: {
            chainNamespace: CHAIN_NAMESPACES.EIP155,
            chainId: "0x5",
          },
        });
        const openloginAdapter = new OpenloginAdapter({
          adapterSettings: {
            network: "mainnet",
            uxMode: "popup",
            loginConfig: {
              jwt: {
                verifier: "web3auth-auth0-demo",
                typeOfLogin: "jwt",
                clientId: "294QRkchfq2YaXUbPri7D6PH7xzHgQMT",
              },
            },
          },
        });
        // web3auth.configureAdapter(openloginAdapter);
        web3auth.configureAdapter(openloginAdapter)
        setWeb3auth(web3auth);
        await web3auth.init();
        if (web3auth.provider) {
          setProvider(web3auth.provider);
        }
      } catch (error) {
        console.error(error);
      }
    };

    init();
  }, []);

  const login = async () => {
    try {
      if (!web3auth) {
        console.log("web3auth not initialized yet");
        return;
      }

      const web3authProvider = await
web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
        loginProvider: "jwt",
        extraLoginOptions: {
          domain: "https://shahbaz-torus.us.auth0.com", // Please append
"https://" before your domain
          verifierIdField: "sub",
        },
      });
      setProvider(web3authProvider);
    } catch (error) {
      console.log(error);
    }

  };




  return (
    <nav className="navbar navbar-expand-lg navbar-light fixed-top py-4
d-block" data-navbar-on-scroll="data-navbar-on-scroll">
      <div className="container">
          <div className="StaticMenu">
            <>
              <button onClick={login}>Login</button>
            </>
          </div>

              <form className="ps-lg-5"><div className="btn btn-light
order-1 order-lg-0">login
                <svg className="bi bi-person-fill" xmlns="
http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor"
viewBox="0 0 16 16">
                  <path d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3
3 0 1 0 0-6 3 3 0 0 0 0 6z"></path>
                </svg></div></form>
            </Link>
          }
        </div>
      </div>
    </nav>
  );
};
export default StaticMenu;

**Comment by sundeep138** [comment]
Also we are using https://shahbaz-torus.us.auth0. and  SVG namespace
(w3.org) <https://www.w3.org/2000/svg>

Thanks
Sundeep

**Comment by sundeep138** [comment]
We tried the following
npm install --save-dev @types/web3auth__core
npm ERR! 404 Not Found - GET ***@***.***%2fweb3-auth
- Not found npm ERR! 404 npm ERR! 404 ***@***.***/web3-auth@*' is not in this
registry. npm ERR! 404 npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url. npm ERR! A complete log
of this run can be found in: npm ERR!
C:\Users\sunde\AppData\Local\npm-cache\_logs\2023-03-09T07_44_47_313Z-debug-0.log

Thanks
Sundeep

**Comment by sundeep138** [comment]
Why you guys are not responding.?
